frontend/src/container/PipelinePage/ListOfPipelines.tsx

```diff
@@ -0,0 +1,361 @@
+			key: e.key,
+			pipelineName: e.pname,
+			filter: e.filter,
+			tags: e.Tags,
+			lastEdited: e.last_edited,
+			editedBy: e.EditedBy,
+			description: e.description,
+		}));
+		setDataSource(updatedData);
+	}, [newAddProcessor]);
+
+	const WarningMessageModal = useCallback(
+		({
+			title,
+			descrition,
+			buttontext,
+			onCancelClick,
+			onOkClick,
+		}: {
+			title: string;
+			descrition: string;
+			buttontext: string;
+			onCancelClick?: () => void;
+			onOkClick?: () => void;
+		}) => {
+			modal.confirm({
+				title: (
+					<Typography.Title level={1} style={modalTitleStyle}>
+						{title}
+					</Typography.Title>
+				),
+				icon: <ExclamationCircleOutlined />,
+				content: <Text style={deleteModalDescriptionStyle}>{descrition}</Text>,
+				okText: <Text>{buttontext}</Text>,
+				cancelText: <Text>Cancel</Text>,
+				onOk: onOkClick,
+				onCancel: onCancelClick,
+				// centered: true,
+			});
+		},
+		[Text, modal],
+	);
+
+	const columns: ColumnsType<PipelineColumn> = [
+		{
+			title: '',
+			dataIndex: 'id',
+			key: 'id',
+			render: (i: number): JSX.Element => (
+				<div>
+					<Avatar style={listIconStyle} size="small">
+						{i}
+					</Avatar>
+				</div>
+			),
+		},
+		{
+			title: 'Pipeline Name',
+			dataIndex: 'pipelineName',
+			key: 'pipelineName',
+		},
+		{
+			title: 'Filters',
+			dataIndex: 'filter',
+			key: 'filter',
+		},
+		{
+			title: 'Tags',
+			dataIndex: 'tags',
+			key: 'tags',
+			render: (value): React.ReactNode =>
+				value?.map((tag: string) => (
+					<Tag color="blue" key={tag}>
+						{tag}
+					</Tag>
+				)),
+		},
+		{
+			title: 'Last Edited',
+			dataIndex: 'lastEdited',
+			key: 'lastEdited',
+		},
+		{
+			title: 'Edited By',
+			dataIndex: 'editedBy',
+			key: 'editedBy',
+		},
+	];
+
+	if (action) {
+		columns.push({
+			title: 'Action',
+			dataIndex: 'action',
+			key: 'action',
+			align: 'center',
+			render: (_value, record): JSX.Element => (
+				<Space size="middle">
+					<span>
+						<EditOutlined style={iconStyle} />
+					</span>
+					<span>
+						<EyeFilled style={iconStyle} />
+					</span>
+					<span>
+						<DeleteFilled
+							onClick={(): void =>
+								WarningMessageModal({
+									title: `Do you want to delete pipeline : ${record.pipelineName}?`,
+									descrition:
+										'Logs are processed sequentially in processors and pipelines. Deleting a pipeline may change content of data processed by other pipelines & processors',
+									buttontext: 'Delete',
+								})
+							}
+							style={iconStyle}
+						/>
+					</span>
+				</Space>
+			),
+		});
+	}
+
+	columns.push({
+		title: '',
+		dataIndex: 'action',
+		key: 'action',
+		render: (): JSX.Element => (
+			<div style={{ display: 'flex', justifyContent: 'center', gap: '20px' }}>
+				<span>
+					<Switch />
+				</span>
+				<span style={{ cursor: 'move' }}>
+					<HolderOutlined style={iconStyle} />
+				</span>
+			</div>
+		),
+	});
+
+	const components = {
```

Comment:
can we make a common components ? same for the draggable columns ?

Reply:
@palashgdev is this new requirement or for future reference?